src/pseudopeople/interface.py

```diff
@@ -115,16 +115,16 @@ def _load_data_from_path(data_path: Path, user_filters: List[Tuple]):
 
 
 def _reformat_dates_for_noising(data: pd.DataFrame, dataset: Dataset):
-    """Formats SSA event_date and dates of birth, so they can be noised."""
+    """Formats date columns so they can be noised as strings."""
     data = data.copy()
-    if COLUMNS.ssa_event_date.name in data.columns and dataset == DATASETS.ssa:
-        # event_date -> YYYYMMDD
-        data[COLUMNS.ssa_event_date.name] = data[COLUMNS.ssa_event_date.name].dt.strftime(
-            "%Y%m%d"
-        )
-    if COLUMNS.dob.name in data.columns:
-        # date_of_birth -> MM/DD/YYYY
-        data[COLUMNS.dob.name] = data[COLUMNS.dob.name].dt.strftime("%m/%d/%Y")
+
+    for date_column in [COLUMNS.dob.name, COLUMNS.ssa_event_date.name]:
+        # Format both the actual column, and the shadow version that will be used
+        # to copy from a household member
+        for column in [date_column, COPY_HOUSEHOLD_MEMBER_COLS.get(date_column)]:
```

Comment:
This fixes an unrelated bug I ran into: when a copy-within-household was performed, the column would become partially strings (original values) and partially true datetimes (copied values). This was silently failing in the swap month and day noise function.